lib/ae_mdw/db/sync/name.ex

```diff
@@ -66,28 +66,31 @@ defmodule AeMdw.Db.Sync.Name do
     m_name_exp = Model.expiration(index: {new_expire, plain_name})
     m_name = Model.name(m_name, expire: new_expire, updates: updates)
 
-    cache_through_delete(Model.ActiveNameExpiration, {old_expire, plain_name})
-
     for ptr <- pointers do
       m_pointee = Model.pointee(index: pointee_key(ptr, {bi, txi}))
       cache_through_write(Model.Pointee, m_pointee)
     end
 
     cond do
       delta_ttl > 0 ->
+        cache_through_delete(Model.ActiveNameExpiration, {old_expire, plain_name})
         cache_through_write(Model.ActiveNameExpiration, m_name_exp)
         cache_through_write(Model.ActiveName, m_name)
 
         log_name_change(height, plain_name, "activate")
 
-      delta_ttl == 0 ->
+      delta_ttl == 0 and not internal? ->
         owner = Model.name(m_name, :owner)
         cache_through_delete(Model.ActiveName, plain_name)
         cache_through_delete(Model.ActiveNameOwner, {owner, plain_name})
+        cache_through_delete(Model.ActiveNameExpiration, {old_expire, plain_name})
```

Comment:
moved here to avoid any expiration operation on the true case (`internal?` or invalid `delta_ttl`)